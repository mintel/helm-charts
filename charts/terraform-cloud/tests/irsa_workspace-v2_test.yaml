suite: Test IRSA Workspaces
templates:
  - irsa-workspace.yaml
release:
  namespace: test-namespace
tests:
  - it: IRSA created for Dev S3 module workspace
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      s3.enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Workspace
        documentIndex: 0
      - equal:
          path: metadata.namespace
          value: test-namespace
        documentIndex: 0
      - equal:
          path: metadata.name
          value: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
        documentIndex: 0
      - hasDocuments:
          count: 2
      - equal:
          path: metadata.annotations["app.mintel.com/altManifestFileSuffix"]
          value: test-app-irsa
        documentIndex: 0
      - equal:
          path: metadata.annotations["argocd.argoproj.io/sync-wave"]
          value: "-20"
        documentIndex: 0
      - contains:
          path: spec.runTriggers
          content:
            name: dev-eu-west-1-cluster1-test-namespace-mntl-test-app-s3
        documentIndex: 0
      - contains:
          path: spec.terraformVariables
          content:
              hcl: false
              name: name
              sensitive: false
              value: test-app
        documentIndex: 0
      - contains:
          path: spec.terraformVariables
          content:
            hcl: false
            name: tfcloud_agent
            sensitive: false
            value: "true"
        documentIndex: 0
      - contains:
          path: spec.terraformVariables
          content:
            hcl: false
            name: eks_cluster_name
            sensitive: false
            value: cluster1
        documentIndex: 0
      - contains:
          path: spec.terraformVariables
          content:
            hcl: false
            name: aws_region
            sensitive: false
            value: eu-west-1
        documentIndex: 0
      - contains:
          path: spec.terraformVariables
          content:
            hcl: false
            name: aws_account_name
            sensitive: false
            value: dev
        documentIndex: 0
      - contains:
          path: spec.terraformVariables
          content:
            hcl: true
            name: tags
            sensitive: false
            value: |-
                {
                  Application = "test-app"
                  Component = "test-app"
                  Owner = "sre"
                  Project = "test-project"
                }
        documentIndex: 0
  - it: IRSA created explicitly
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      irsa.enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Workspace
        documentIndex: 0
      - equal:
          path: metadata.namespace
          value: test-namespace
        documentIndex: 0
      - equal:
          path: metadata.name
          value: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: IRSA created with modified syncWave
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      irsa.enabled: true
      irsa.syncWave: "-100"
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Workspace
        documentIndex: 0
      - equal:
          path: metadata.namespace
          value: test-namespace
        documentIndex: 0
      - equal:
          path: metadata.name
          value: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
        documentIndex: 0
      - hasDocuments:
          count: 2
      - equal:
          path: metadata.annotations["argocd.argoproj.io/sync-wave"]
          value: "-100"
        documentIndex: 0
  - it: IRSA created with default (0) syncWave
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      irsa.enabled: true
      irsa.syncWave: "0"
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: Workspace
        documentIndex: 0
      - equal:
          path: metadata.namespace
          value: test-namespace
        documentIndex: 0
      - equal:
          path: metadata.name
          value: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
        documentIndex: 0
      - hasDocuments:
          count: 2
      - equal:
          path: metadata.annotations["argocd.argoproj.io/sync-wave"]
          value: "0"
        documentIndex: 0
