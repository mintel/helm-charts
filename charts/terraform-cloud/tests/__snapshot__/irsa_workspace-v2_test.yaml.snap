IRSA created explicitly:
  1: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPool:
        id: ""
      allowDestroyPlan: true
      applyMethod: auto
      executionMode: agent
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      notifications:
        - enabled: true
          name: tfcloud-auto-approver
          token: ${TFCLOUD_AUTO_APPROVER_SIGNATURE_KEY}
          triggers:
            - run:needs_attention
          type: generic
          url: ${TFCLOUD_AUTO_APPROVER_URL}
      organization: Mintel
      runTriggers: null
      sshKey:
        name: mintel-ssh
      terraformVariables:
        - hcl: false
          name: aws_account_name
          sensitive: false
          value: dev
        - hcl: false
          name: aws_region
          sensitive: false
          value: eu-west-1
        - hcl: false
          name: eks_cluster_name
          sensitive: false
          value: cluster1
        - hcl: false
          name: k8s_namespace
          sensitive: false
          value: test-namespace
        - hcl: false
          name: k8s_service_account_name
          sensitive: false
          value: test-app
        - hcl: false
          name: name
          sensitive: false
          value: test-app
        - hcl: true
          name: tags
          sensitive: false
          value: |-
            {
              Application = "test-app"
              Component = "test-app"
              Owner = "sre"
              Project = "test-project"
            }
        - hcl: false
          name: tfcloud_agent
          sensitive: false
          value: "true"
      terraformVersion: 1.3.10
      token:
        secretKeyRef:
          key: token
          name: terraformrc
  2: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Module
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      destroyOnDeletion: true
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.5.1
      name: operator
      organization: Mintel
      restartedAt: 77a938f5e67a5feb8645c0a50b19f7d68df333783aaca14c710f173c1f1d88bf
      token:
        secretKeyRef:
          key: token
          name: terraformrc
      variables:
        - name: aws_account_name
        - name: aws_region
        - name: eks_cluster_name
        - name: k8s_namespace
        - name: k8s_service_account_name
        - name: name
        - name: tags
        - name: tfcloud_agent
      workspace:
        name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
IRSA created for Dev S3 module workspace:
  1: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPool:
        id: ""
      allowDestroyPlan: true
      applyMethod: auto
      executionMode: agent
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      notifications:
        - enabled: true
          name: tfcloud-auto-approver
          token: ${TFCLOUD_AUTO_APPROVER_SIGNATURE_KEY}
          triggers:
            - run:needs_attention
          type: generic
          url: ${TFCLOUD_AUTO_APPROVER_URL}
      organization: Mintel
      runTriggers:
        - name: dev-eu-west-1-cluster1-test-namespace-mntl-test-app-s3
      sshKey:
        name: mintel-ssh
      terraformVariables:
        - hcl: false
          name: aws_account_name
          sensitive: false
          value: dev
        - hcl: false
          name: aws_region
          sensitive: false
          value: eu-west-1
        - hcl: false
          name: eks_cluster_name
          sensitive: false
          value: cluster1
        - hcl: false
          name: k8s_namespace
          sensitive: false
          value: test-namespace
        - hcl: false
          name: k8s_service_account_name
          sensitive: false
          value: test-app
        - hcl: false
          name: name
          sensitive: false
          value: test-app
        - hcl: true
          name: tags
          sensitive: false
          value: |-
            {
              Application = "test-app"
              Component = "test-app"
              Owner = "sre"
              Project = "test-project"
            }
        - hcl: false
          name: tfcloud_agent
          sensitive: false
          value: "true"
      terraformVersion: 1.3.10
      token:
        secretKeyRef:
          key: token
          name: terraformrc
  2: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Module
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      destroyOnDeletion: true
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.5.1
      name: operator
      organization: Mintel
      restartedAt: 77a938f5e67a5feb8645c0a50b19f7d68df333783aaca14c710f173c1f1d88bf
      token:
        secretKeyRef:
          key: token
          name: terraformrc
      variables:
        - name: aws_account_name
        - name: aws_region
        - name: eks_cluster_name
        - name: k8s_namespace
        - name: k8s_service_account_name
        - name: name
        - name: tags
        - name: tfcloud_agent
      workspace:
        name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
IRSA created with default (0) syncWave:
  1: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "0"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPool:
        id: ""
      allowDestroyPlan: true
      applyMethod: auto
      executionMode: agent
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      notifications:
        - enabled: true
          name: tfcloud-auto-approver
          token: ${TFCLOUD_AUTO_APPROVER_SIGNATURE_KEY}
          triggers:
            - run:needs_attention
          type: generic
          url: ${TFCLOUD_AUTO_APPROVER_URL}
      organization: Mintel
      runTriggers: null
      sshKey:
        name: mintel-ssh
      terraformVariables:
        - hcl: false
          name: aws_account_name
          sensitive: false
          value: dev
        - hcl: false
          name: aws_region
          sensitive: false
          value: eu-west-1
        - hcl: false
          name: eks_cluster_name
          sensitive: false
          value: cluster1
        - hcl: false
          name: k8s_namespace
          sensitive: false
          value: test-namespace
        - hcl: false
          name: k8s_service_account_name
          sensitive: false
          value: test-app
        - hcl: false
          name: name
          sensitive: false
          value: test-app
        - hcl: true
          name: tags
          sensitive: false
          value: |-
            {
              Application = "test-app"
              Component = "test-app"
              Owner = "sre"
              Project = "test-project"
            }
        - hcl: false
          name: tfcloud_agent
          sensitive: false
          value: "true"
      terraformVersion: 1.3.10
      token:
        secretKeyRef:
          key: token
          name: terraformrc
  2: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Module
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "0"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      destroyOnDeletion: true
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.5.1
      name: operator
      organization: Mintel
      restartedAt: 77a938f5e67a5feb8645c0a50b19f7d68df333783aaca14c710f173c1f1d88bf
      token:
        secretKeyRef:
          key: token
          name: terraformrc
      variables:
        - name: aws_account_name
        - name: aws_region
        - name: eks_cluster_name
        - name: k8s_namespace
        - name: k8s_service_account_name
        - name: name
        - name: tags
        - name: tfcloud_agent
      workspace:
        name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
IRSA created with modified syncWave:
  1: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-100"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPool:
        id: ""
      allowDestroyPlan: true
      applyMethod: auto
      executionMode: agent
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      notifications:
        - enabled: true
          name: tfcloud-auto-approver
          token: ${TFCLOUD_AUTO_APPROVER_SIGNATURE_KEY}
          triggers:
            - run:needs_attention
          type: generic
          url: ${TFCLOUD_AUTO_APPROVER_URL}
      organization: Mintel
      runTriggers: null
      sshKey:
        name: mintel-ssh
      terraformVariables:
        - hcl: false
          name: aws_account_name
          sensitive: false
          value: dev
        - hcl: false
          name: aws_region
          sensitive: false
          value: eu-west-1
        - hcl: false
          name: eks_cluster_name
          sensitive: false
          value: cluster1
        - hcl: false
          name: k8s_namespace
          sensitive: false
          value: test-namespace
        - hcl: false
          name: k8s_service_account_name
          sensitive: false
          value: test-app
        - hcl: false
          name: name
          sensitive: false
          value: test-app
        - hcl: true
          name: tags
          sensitive: false
          value: |-
            {
              Application = "test-app"
              Component = "test-app"
              Owner = "sre"
              Project = "test-project"
            }
        - hcl: false
          name: tfcloud_agent
          sensitive: false
          value: "true"
      terraformVersion: 1.3.10
      token:
        secretKeyRef:
          key: token
          name: terraformrc
  2: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Module
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-100"
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      destroyOnDeletion: true
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.5.1
      name: operator
      organization: Mintel
      restartedAt: 77a938f5e67a5feb8645c0a50b19f7d68df333783aaca14c710f173c1f1d88bf
      token:
        secretKeyRef:
          key: token
          name: terraformrc
      variables:
        - name: aws_account_name
        - name: aws_region
        - name: eks_cluster_name
        - name: k8s_namespace
        - name: k8s_service_account_name
        - name: name
        - name: tags
        - name: tfcloud_agent
      workspace:
        name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
